packages/headless/src/controllers/facets/category-facet/headless-category-facet-options.ts

```diff
@@ -17,14 +15,79 @@ import {
   facetSearch,
 } from '../_common/facet-option-definitions';
 
-type OptionalFacetId = Partial<
-  Pick<CategoryFacetRegistrationOptions, 'facetId'>
->;
+export interface CategoryFacetOptions {
+  /**
+   * The field whose values you want to display in the facet.
+   * */
+  field: string;
+  /**
+   * The character that specifies the hierarchical dependency.
+   * @default ">"
+   */
+  delimitingCharacter?: string;
+  /**
+   * A unique identifier for the controller. By default, a unique random identifier is generated.
+   * */
+  facetId?: string;
+  /**
+   * Facet search options.
+   */
+  facetSearch?: CategoryFacetSearchOptions;
+  /**
+   * Whether to exclude folded result parents when estimating the result count for each facet value.
+   * @defaut true
+   */
+  filterFacetCount?: boolean;
+  /**
+   * The maximum number of results to scan in the index to ensure that the facet lists all potential facet values.
+   *
+   * Note: A high injectionDepth may negatively impact the facet request performance.
+   *
+   * @default 1000
+   * @minimum 0
+   * */
+  injectionDepth?: number;
+  /**
+   * The number of values to request for this facet. Also determines the number of additional values to request each time this facet is expanded, and the number of values to display when this facet is collapsed.
+   *
+   * @default 8
+   * @minimum 1
+   */
+  numberOfValues?: number;
+  /**
+   * The base path shared by all values for the facet.
+   * @default []
+   */
+  basePath?: string[];
+  /**
+   * Whether to use basePath as a filter for the results.
+   * @default true
+   */
+  filterByBasePath?: boolean;
```

Comment:
In addition to grouping the options as `required` first and `optional` second, a nice touch would be to sort the options alphabetically within each group similar to VSCode.

e.g.

```
field: string;
basePath?: string[];
...
```